lldb/packages/Python/lldbsuite/test/lldbtest.py

```diff
@@ -1799,6 +1799,7 @@ def no_reason(_):
                 for cat in categories:
 
                     @decorators.add_test_categories([cat])
+                    @decorators.remove_test_categories(categories)
```

Comment:
I don't understand how this loop works. for category in categories, add the category, then remove all categories?

Or is all this delayed until the decorator is invoked later?

Reply:
(Also, the decorators are applied from the inside out)